### PR TITLE
Find quote/citation with gnarly regex

### LINKED ISSUE #573: Alternative quote style
To accommodate recent articles which feature a quote directly after the stand-first, we need to add a new quote style which can only be used in this particular area of an article.

Text size is same as normal body copy. Indent text 30px from the left and add larger quote mark (Helvetica). Quote reference name is same style as other quote styles

screen shot 2017-02-17 at 14 51 45

### PR DESCRIPTION
**Pull request by davidpmccormick**
## What is this PR trying to achieve?
Adding styling to `blockquote`s we get from WP. If the `blockquote` contains curly quotes, then we pluck everything out of them and use it as the body of the quote. If there's anything else left over, we treat that as the `cite`. If it doesn't contain curly quotes, then all of the content inside the `blockquote` gets used as the body of the quote.

Have checked this on a large sample of our existing articles – believe edge cases have been ticked off.

Note: all pre-existing styling _within_ the cite block will be removed.

Fixes #573.

## What does it look like?
screen shot 2017-04-12 at 14 18 48

### CODE REVIEW

File: server/util/body-parser.js [hunk 1]
```diff
@@ -111,6 +112,56 @@ export function convertWpHeading(node) {
...
+
+function splitBlockquote(blockquote) {
+  // Match a string within quote marks and capture possible non-<p> wrapping tags
+  // Visualisation: https://regexper.com/#(%3F%3A%3C(%3F!p)(.*%3F)%3E)%3F%E2%80%9C%5B%5Cs%5CS%5D*%3F%E2%80%9D(%3F%3A.*%3C%5C%2F(%3F!p)(.*%3F)%3E)%3F
+  const quoteAndTagsRegex = new RegExp(/(?:<(?!p)(.*?)>)?“[\s\S]*?”(?:.*<\/(?!p)(.*?)>)?/);
+  const quoteAndTags = blockquote.match(quoteAndTagsRegex);
+
+  if (quoteAndTags) {
+    const wrapper = getWrappingTags(quoteAndTags);
+    const quote = blockquote.match(quoteAndTagsRegex)[0].replace(/[“”]/g, '');
```
Comment (jamesgorrie) [inline server/util/body-parser.js]: We do a lot of accessing array elements by key:
https://github.com/wellcometrust/wellcomecollection.org/pull/784/files#diff-6ddd737b5aca2e7ffcb9d0c91b5c0960R116

https://github.com/wellcometrust/wellcomecollection.org/pull/784/files#diff-6ddd737b5aca2e7ffcb9d0c91b5c0960R117

Could we potentially wrap this in a try / catch - or are we confident that we won't break ever?

File: server/views/components/article-body/index.njk [hunk 2]
```diff
@@ -25,6 +25,8 @@
         {% component 'tweet', { tweet: bodyPart.value } %}
       {% elif bodyPart.type === 'instagramEmbed' %}
         {% component 'instagram-embed', { instagramEmbed: bodyPart.value } %}
+      {% elif bodyPart.type === 'blockquote' %}
+        {% component 'blockquote', { body: bodyPart.value.body | safe, footer: bodyPart.value.cite | safe, modifiers: ['small'] } %}
```
Comment (jamesgorrie) [inline server/views/components/article-body/index.njk]: At some stage we need to fix cardigan to say this is the blockquote, and the other is the pull quote.

File: server/util/body-parser.js (hunk 1 above)
Comment (davidpmccormick) [inline server/util/body-parser.js]: We're inside the `if (quoteAndTags)` conditional here, so accessing by key is safe. And we know that the Regex has two matching groups. So once we've checked that they're not undefined and they are equal, we're good.

File: server/views/components/article-body/index.njk (hunk 2 above)
Comment (davidpmccormick) [inline server/views/components/article-body/index.njk]: There are two other variants currently in Cardigan. I'm unsure of their originally intended uses  – will check with @Norvard if they're both needed.

Comment (jennpb) [inline server/views/components/article-body/index.njk]: We will want to keep both styles.
Blockquote forms part of the narrative (often preceded by a colon)
Pullquote is an extract from the narrative

Comment (davidpmccormick) [inline server/views/components/article-body/index.njk]: What's the difference between the two other variants?

Comment (davidpmccormick) [inline server/views/components/article-body/index.njk]: Actually, we've got [four variants](https://cardigan.wellcomecollection.org/components/detail/blockquote--small.html) now, with `--small` being the one we're using for the WP blockquotes

Comment (jamesgorrie) [inline server/views/components/article-body/index.njk]: `--small` = `blockquote`.
The other ones we don't use as yet - although I think we agreed that the normal / large one looked like a pullquote.

**Pull request merged**